pkg/controller/clusterautoscaler/clusterautoscaler_controller.go

```diff
@@ -17,43 +17,76 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 const (
-	caImage          = "quay.io/bison/cluster-autoscaler:a554b4f5"
 	criticalPod      = "scheduler.alpha.kubernetes.io/critical-pod"
 	caServiceAccount = "cluster-autoscaler"
 )
 
 // Add creates a new ClusterAutoscaler Controller and adds it to the
 // Manager. The Manager will set fields on the Controller and Start it
 // when the Manager is Started.
-func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+func Add(mgr manager.Manager, cfg *operator.Config) error {
+	r := newReconciler(mgr)
+
+	r.SetImage(cfg.ClusterAutoscalerImage)
+	r.SetReplicas(cfg.ClusterAutoscalerReplicas)
+	r.SetNamespace(cfg.ClusterAutoscalerNamespace)
+
+	return add(mgr, cfg, r)
 }
 
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+// newReconciler returns a new Reconciler.
+func newReconciler(mgr manager.Manager) *Reconciler {
 	return &Reconciler{
 		client: mgr.GetClient(),
 		scheme: mgr.GetScheme(),
 	}
 }
 
+// processEvent is used in predicate functions.  It inspects metadata and
+// returns a boolean value representing whether the controller should process an
+// event for the object.
+func processEvent(meta metav1.Object, cfg *operator.Config) bool {
+	// Only process events for objects matching the configured resource name.
+	if meta.GetName() != cfg.ClusterAutoscalerName {
+		glog.Warningf("Not processing ClusterAutoscaler %s", meta.GetName())
```

Comment:
We should probably indicate that only `config.ClusterAutoscalerName` will be watched

Reply:
I wonder if we could somehow set the name programatically so no need to arbitrarily having to match the manifests and the config names, I don't have any good suggestion atm though

Reply:
unit test for `processEvent` would be nice

Reply:
I don't think we can do anything other than filter on names like this. This is honestly pretty weird. These kinds of singleton resources that are managing applications that are *already* native Kubernetes apps have dubious benefits. In a perfect world, the cluster-autoscaler itself would just be doing this. There would be no need for an operator.